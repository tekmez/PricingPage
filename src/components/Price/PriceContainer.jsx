import React from 'react'
import { Text, View } from 'react-native'
import PriceBadge from '../PriceBadge'
import styles from './Price.style'

const PriceContainer = ({ saving, title, price, boldPrice }) => {
  return (
    <View>
      {saving && <PriceBadge title={saving} />}
      <Text testID="price-container-title" style={styles.title}>
        {title}
      </Text>
      <View style={styles.priceContainer}>
        {price && (
          <Text testID="light-price-title" style={styles.lightPrice}>
            {price}
          </Text>
        )}
        <Text testID="bold-price-title" style={styles.boldPrice}>
          {boldPrice}
        </Text>
      </View>
    </View>
  )
}

export default PriceContainer
