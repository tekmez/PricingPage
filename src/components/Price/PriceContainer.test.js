import React from 'react'
import { render } from '@testing-library/react-native'
import PriceContainer from './PriceContainer'

describe('Price Container', () => {
  it('should match snapshot', () => {
    const comp = render(<PriceContainer />).toJSON()
    expect(comp).toMatchSnapshot()
  })

  it('should render correctly without saving', () => {
    const title = 'Price Container'
    const price = '4.00'
    const boldPrice = '5.00'
    const comp = render(
      <PriceContainer title={title} price={price} boldPrice={boldPrice} />
    )

    const priceTitle = comp.getByTestId('price-container-title').children[0]
    const priceText = comp.getByTestId('light-price-title').children[0]
    const boldPriceText = comp.getByTestId('bold-price-title').children[0]

    expect(priceTitle).toBe(title)
    expect(priceText).toBe(price)
    expect(boldPriceText).toBe(boldPrice)
  })

  it('should render correctly with saving', () => {
    const title = 'Price Container'
    const price = '4.00'
    const boldPrice = '5.00'
    const saving = '1.00'
    const comp = render(
      <PriceContainer
        title={title}
        price={price}
        boldPrice={boldPrice}
        saving={saving}
      />
    )
    const priceTitle = comp.getByTestId('price-title').children[0]
    expect(priceTitle).toBe(saving)
  })
})
