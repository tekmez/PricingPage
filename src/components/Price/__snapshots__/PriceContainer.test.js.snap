// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Price Container should match snapshot 1`] = `
<View>
  <Text
    style={
      {
        "color": "#717578",
        "fontFamily": "Inter-Regular",
        "fontSize": 12,
        "letterSpacing": 0.5,
        "lineHeight": 16,
      }
    }
    testID="price-container-title"
  />
  <View
    style={
      {
        "flexDirection": "row",
        "gap": 6,
      }
    }
  >
    <Text
      style={
        {
          "fontFamily": "Inter-Bold",
          "fontSize": 12,
          "fontWeight": "600",
          "lineHeight": 16,
        }
      }
      testID="bold-price-title"
    />
  </View>
</View>
`;
